lib/file-source/src/file_watcher.rs

```diff
@@ -152,6 +163,26 @@ impl FileWatcher {
             }
         }
     }
+
+    fn track_read_attempt(&mut self) {
+        self.last_read_attempt = Some(Instant::now());
+    }
+
+    fn track_read_success(&mut self) {
+        self.last_read_success = Some(Instant::now());
+    }
+
+    pub fn should_read(&self) -> bool {
+        match (self.last_read_attempt, self.last_read_success) {
+            (None, None) => true,
+            (None, Some(_)) => panic!("success requires an attempt"),
```

Comment:
```suggestion
            (None, Some(_)) => unreachable!("success requires an attempt"),
```

do we want to use this here?

Reply:
I actually simplified this away, sorry :smile: